pennylane/vqe/vqe.py

```diff
@@ -370,70 +370,145 @@ class VQECost:
             Supports all interfaces supported by the :func:`~.qnode` decorator.
         diff_method (str, None): The method of differentiation to use with the created cost function.
             Supports all differentiation methods supported by the :func:`~.qnode` decorator.
+        optimize (bool): Whether to optimize the observables composing the Hamiltonian by
+            separating them into qubit-wise commuting groups. Each group can then be executed
+            within a single QNode, resulting in fewer QNodes to evaluate.
 
     Returns:
         callable: a cost function with signature ``cost_fn(params, **kwargs)`` that evaluates
         the expectation of the Hamiltonian on the provided device(s)
 
     .. seealso:: :class:`~.Hamiltonian`, :func:`~.generate_hamiltonian`, :func:`~.map`, :func:`~.dot`
 
-    **Example:**
+    .. UsageDetails::
```

Comment:
I'm personally not such a big fan of hiding _everything_ behind 'Usage Details'. Is there a minimal example we can retain above (in a standard `**Example**` section), and keep 'Usage Details' for more advanced usage examples?

Reply:
The minimal example could just be 'this is how VQECost is used and evaluated' in a couple of lines

Reply:
```rest
**Example**

To construct a ``VQECost`` cost function, we require a Hamiltonian
to measure, and an ansatz for our variational circuit.

We can construct a Hamiltonian manually,

.. code-block:: python

    coeffs = [0.2, -0.543]
    obs = [
        qml.PauliX(0) @ qml.PauliZ(1) @ qml.PauliY(3),
        qml.PauliZ(0) @ qml.Hadamard(2)
    ]
    H = qml.vqe.Hamiltonian(coeffs, obs)

or, alternatively, the :func:`~.generate_hamiltonian` function from the
:doc:`/introduction/chemistry` module can be used to generate a molecular
Hamiltonian.

Once we have our Hamiltonian, we can select an ansatz and construct
the cost function.

>>> ansatz = qml.templates.StronglyEntanglingLayers
>>> cost = qml.VQECost(ansatz, H, dev, interface="torch")
>>> params = torch.rand([2, 2, 3])
>>> cost(params)
tensor(0.0245, dtype=torch.float64)
```

Reply:
Using the function ``molecular_hamiltonian`` might help here:

```
>>> h, qubits = qml.qchem.molecular_hamiltonian('h2','h2.xyz')
>>> ansatz = qml.templates.StronglyEntanglingLayers
>>> params = torch.rand([2, 4, 3])
UPDATE OUTPUT
```

Reply:
Thanks! Done!